### PR TITLE
Computation of the Chebyshev coefficients

### PR DESCRIPTION
**Pull request by sofiachiarenza**
Hi @marcobonici, the functions pass the CI and I checked that with the N5K interpolated power spectrum the coefficients are exactly the same (sum(my_coeff .- check_coeff) = 0.0). Should I add this as a test? Or is that too specific?

### CODE REVIEW

File: src/chebcoefs.jl [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+# Returns
+- `p::FFTW.rFFTWPlan`: An FFTW plan object for transforming `vals` with the appropriate R2R transformations. This plan can be applied using the `*` operator (e.g., `transformed_vals = p * vals`).
+
+"""
+function plan_fft(vals::AbstractArray{<:Number,N}) where {N}
+    kind = map(n -> n > 1 ? FFTW.REDFT00 : FFTW.DHT, size(vals)[1])
+    p = FFTW.plan_r2r(deepcopy(vals), kind, [1]; flags=FFTW.PATIENT, timelimit=Inf)   
+                                                                                    
+    return p 
+end
```
Comment (marcobonici) [inline src/chebcoefs.jl:19]: We will focus on this in a later stage, but we should **specialize** the code here.
In particular, we know that the input array has two axes.
For now, this is fine.

File: src/chebcoefs.jl [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+   plan_fft(vals::AbstractArray{<:Number,N})
+
+Create an FFTW real-to-real (R2R) transformation plan for the first axis of a given multidimensional array `vals`. 
+In practise, the `vals` array is the power spectrum P(k,χ). The first axis should then contain the wavenumbers `k`.
+
+# Arguments
+- `vals::AbstractArray{<:Number, N}`: The input array of any numerical type with `N` dimensions.
+
+# Returns
+- `p::FFTW.rFFTWPlan`: An FFTW plan object for transforming `vals` with the appropriate R2R transformations. This plan can be applied using the `*` operator (e.g., `transformed_vals = p * vals`).
```
Comment (marcobonici) [inline src/chebcoefs.jl:11]: Make more explicit the R2R. It should be clear from the context, but explicit is better than implicit.
```suggestion
- `p::FFTW.rFFTWPlan`: An FFTW plan object for transforming `vals` with the appropriate real to real transformations. This plan can be applied using the `*` operator (e.g., `transformed_vals = p * vals`).
```

File: src/chebcoefs.jl [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    fast_chebcoefs(vals::AbstractArray{<:Number,N})
+
+Efficiently compute the Chebyshev coefficients of a multidimensional array `vals` using an O(n log n) method. This method leverages FFT-based type-I Discrete Cosine Transform (DCT-I).
+
+Arguments:
+- `vals::AbstractArray{<:Number,N}`: A multidimensional array of values for which to compute the Chebyshev coefficients.
+
+Returns:
+- `coefs`: An array of the same size as `vals`, containing the computed Chebyshev coefficients.
+"""
```
Comment (marcobonici) [inline src/chebcoefs.jl:32]: Update the docs, since the signature of the function has changed.

**Review by marcobonici — COMMENTED** [review]
Hi @sofiachiarenza , there are a couple of minor things (mostly in the docs) that I am asking you to change. After this, we are good to go!

**Comment by sofiachiarenza** [comment]
@marcobonici done!

**Review by marcobonici — APPROVED** [review]

**Pull request merged**